ci: almalinux build fails due to relro

### What happened?

`g-ir-scanner` is failing in Almalinux packaging builds. I instrumented Meson and it appears that the RPM build is setting the following `LDFLAGS`:

```
LDFLAGS=-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 
```

Either we need to reset LDFLAGS somehow when invoking `g-ir-scanner`, or convince it to build with `-fPIE`. I patched Meson `gnome.py` to experiment with this and it worked, but obviously that's not a full solution

### Stack Trace

```
[18/23] Generating adbc-glib/ADBC-1.0.gir with a custom command (wrapped by meson to set env)
FAILED: adbc-glib/ADBC-1.0.gir 
env PKG_CONFIG_PATH=/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/meson-uninstalled:/usr/lib64/pkgconfig:/usr/share/pkgconfig PKG_CONFIG=/usr/bin/pkg-config CC=gcc CFLAGS= /usr/bin/g-ir-scanner --quiet --no-libtool --namespace=ADBC --nsversion=1.0 --warn-all --output adbc-glib/ADBC-1.0.gir --c-include=adbc-glib/adbc-glib.h --warn-all -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/adbc-glib -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/adbc-glib -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/. -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/. -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/../c/include -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/../c/include --filelist=/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/adbc-glib/libadbc-glib.so.1.6.0.p/ADBC_1.0_gir_filelist --include=GObject-2.0 --symbol-prefix=gadbc --identifier-prefix=GADBC --pkg-export=adbc-glib --cflags-begin -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/. -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/. -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/../c/include -I/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/../c/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/gobject-introspection-1.0 --cflags-end --add-include-path=/usr/share/gir-1.0 -L/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/adbc-glib --library adbc-glib -L/root/rpmbuild/BUILD/apache-arrow-adbc-18/glib/../c/redhat-linux-build/driver_manager --extra-library=gobject-2.0 --extra-library=glib-2.0 --extra-library=girepository-1.0 --sources-top-dirs /build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/ --sources-top-dirs /build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/ --warn-error
/usr/bin/ld: /build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/tmp-introspect4ug84fdq/ADBC-1.0.o: relocation R_X86_64_32 against `.rodata' can not be used when making a PIE object; recompile with -fPIE
collect2: error: ld returned 1 exit status
linking of temporary binary failed: Command '['gcc', '-o', '/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/tmp-introspect4ug84fdq/ADBC-1.0', '/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/tmp-introspect4ug84fdq/ADBC-1.0.o', '-L.', '-Wl,-rpath,.', '-Wl,--no-as-needed', '-L/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/adbc-glib', '-Wl,-rpath,/build/rpmbuild/BUILD/apache-arrow-adbc-18/glib/build/adbc-glib', '-L/root/rpmbuild/BUILD/apache-arrow-adbc-18/glib/../c/redhat-linux-build/driver_manager', '-Wl,-rpath,/root/rpmbuild/BUILD/apache-arrow-adbc-18/glib/../c/redhat-linux-build/driver_manager', '-ladbc-glib', '-lgobject-2.0', '-lglib-2.0', '-lgirepository-1.0', '-lgio-2.0', '-lgobject-2.0', '-Wl,--export-dynamic', '-lgmodule-2.0', '-pthread', '-lglib-2.0', '-lglib-2.0', '-Wl,-z,relro', '-Wl,--as-needed', '-Wl,-z,now', '-specs=/usr/lib/rpm/redhat/redhat-hardened-ld', '-specs=/usr/lib/rpm/redhat/redhat-annobin-cc1']' returned non-zero exit status 1.
ninja: build stopped: subcommand failed.
```

### How can we reproduce the bug?

```
cd ci/linux-packages/
env YUM_TARGETS=almalinux-9 DISTRIBUTION=almalinux VERSION=18 ARROW_SOURCE=/home/lidavidm/Code/arrow rake yum:build BUILD_DIR=build
```

### Environment/Setup

_No response_

--- a/dev/release/.env.example
+++ b/dev/release/.env.example
@@ -30,6 +30,7 @@
 #
 # You must set this.
 #
-# This can be obtained by logging in to https://apache.jfrog.io with
-# your ASF credentials and generating an API key from your user profile.
+# This can be obtained by logging in to
+# https://packages.apache.org/ui/login with your ASF credentials and
+# generating an API key from your user profile.
 #ARTIFACTORY_API_KEY=secret

--- a/dev/release/06-binary-verify.sh
+++ b/dev/release/06-binary-verify.sh
@@ -102,9 +102,9 @@ Note: to verify APT/YUM packages on macOS/AArch64, you must \`export DOCKER_DEFA
 [1]: https://github.com/apache/arrow-adbc/issues?q=is%3Aissue+milestone%3A%22ADBC+Libraries+${RELEASE}%22+is%3Aclosed
 [2]: https://github.com/apache/arrow-adbc/commit/${commit}
 [3]: https://dist.apache.org/repos/dist/dev/arrow/${tag}/
-[4]: https://apache.jfrog.io/artifactory/arrow/almalinux-rc/
-[5]: https://apache.jfrog.io/artifactory/arrow/debian-rc/
-[6]: https://apache.jfrog.io/artifactory/arrow/ubuntu-rc/
+[4]: https://packages.apache.org/artifactory/arrow/almalinux-rc/
+[5]: https://packages.apache.org/artifactory/arrow/debian-rc/
+[6]: https://packages.apache.org/artifactory/arrow/ubuntu-rc/
 [7]: https://repository.apache.org/content/repositories/staging/org/apache/arrow/adbc/
 [8]: https://github.com/apache/arrow-adbc/releases/tag/${tag}
 [9]: https://github.com/apache/arrow-adbc/blob/${tag}/CHANGELOG.md

--- a/dev/release/post-09-announce.sh
+++ b/dev/release/post-09-announce.sh
@@ -68,7 +68,7 @@ The Apache Arrow Community
 
 [1]: https://github.com/apache/arrow-adbc/issues?q=is%3Aissue+milestone%3A%22ADBC+Libraries+${RELEASE}%22+is%3Aclosed
 [2]: https://arrow.apache.org/adbc/current/driver/installation.html
-[3]: https://apache.jfrog.io/ui/native/arrow
+[3]: https://packages.apache.org/ui/native/arrow
 [4]: https://arrow.apache.org/blog/2023/01/05/introducing-arrow-adbc/
 [5]: https://lists.apache.org/list.html?user@arrow.apache.org
 [6]: https://lists.apache.org/list.html?dev@arrow.apache.org

--- a/dev/release/verify-apt.sh
+++ b/dev/release/verify-apt.sh
@@ -67,7 +67,7 @@ ${APT_INSTALL} \
 
 code_name="$(lsb_release --codename --short)"
 distribution="$(lsb_release --id --short | tr 'A-Z' 'a-z')"
-artifactory_base_url="https://apache.jfrog.io/artifactory/arrow/${distribution}"
+artifactory_base_url="https://packages.apache.org/artifactory/arrow/${distribution}"
 case "${TYPE}" in
   rc|staging-rc|staging-release)
     suffix=${TYPE%-release}
@@ -94,7 +94,7 @@ curl \
     --fail \
     --location \
     --remote-name \
-     https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+     https://packages.apache.org/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
 ${APT_INSTALL} ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
 
 if [ "${TYPE}" = "local" ]; then

--- a/dev/release/verify-yum.sh
+++ b/dev/release/verify-yum.sh
@@ -39,7 +39,7 @@ TYPE="$2"
 SOURCE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 local_prefix="${SOURCE_DIR}/../../ci/linux-packages"
 
-artifactory_base_url="https://apache.jfrog.io/artifactory/arrow"
+artifactory_base_url="https://packages.apache.org/artifactory/arrow"
 
 distribution=$(. /etc/os-release && echo "${ID}")
 distribution_version=$(. /etc/os-release && echo "${VERSION_ID}" | grep -o "^[0-9]*")
@@ -94,7 +94,7 @@ fi
 if [ "${TYPE}" = "local" ]; then
   sed \
     -e "s,^\\[apache-arrow-,\\[apache-adbc-,g" \
-    -e "s,baseurl=https://apache\.jfrog\.io/artifactory/arrow/,baseurl=file://${local_prefix}/yum/repositories/,g" \
+    -e "s,baseurl=https://packages\.apache\.org/artifactory/arrow/,baseurl=file://${local_prefix}/yum/repositories/,g" \
     -e "s,RPM-GPG-KEY-Apache-Arrow,RPM-GPG-KEY-Apache-ADBC,g" \
     /etc/yum.repos.d/Apache-Arrow.repo > \
     /etc/yum.repos.d/Apache-ADBC.repo

